File: LeetCode/LeetCode/rotate.cc
#include<iostream>
#include<vector>

using namespace std;

class Solution{
    public:
        void rotate(vector<vector<int>> &matrix) {
            const int n = matrix.size();
            for(int i = i; i < n / 2; i++) {
                for(int j = i; j < n -1 -i; ++j) {
                    const auto tmp = matrix[i][j];
                    matrix[i][j] = matrix[n-1-j][i];
                    matrix[n-1-j][i] = matrix[n-1-i][n-1-j];
                    matrix[n-1-i][n-1-j] = matrix[j][n-1-i];
                    matrix[j][n-1-i] = tmp;
                }
            }
        }
};

File: LeetCode/LeetCode/rotate.py
class Solution:
    def rotate(self, matrix):
        return [list(reversed(x)) for x in zip(*matrix)]


if __name__ == "__main__":
    matrix = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
    print(Solution().rotate(matrix))
